### PR TITLE
feat(sort-enums): handles numeric operations

### LINKED ISSUE #386: Bug: sort-enums negative values not sorted correctly with forceNumericSort: true
### Describe the bug

In the sort-enums rule when the forceNumericSort: true flag is set a negative value causes the ordering to fallback on the configured type (alphabetical etc.)

### Code example

```typescript
export enum NumericTypeCorrectOrder {
    One = 1,
    Two = 2,
    Three = 3
}

export enum NumericTypeWrongOrder {
    One = 1,
    OneNegative = -1, 
    Three = 3,
    Two = 2
}

```

### ESLint version

v9.15.0

### ESLint Plugin Perfectionist version

v4.0.3

### Additional comments

The sort-enums.ts source code has a check for determining an enum is a numeric one (https://github.com/azat-io/eslint-plugin-perfectionist/blob/main/rules/sort-enums.ts#L157)

```typescript
     let isNumericEnum = members.every(
        member =>
          member.initializer?.type === 'Literal' &&
          typeof member.initializer.value === 'number',
      )
```

I locally logged out the type for the members, and when it hit the negative value it had a type of `UnaryExpression` instead of `Literal`

### Validations

- [X] Read the [docs](https://perfectionist.dev).
- [X] Check that there isn't [already an issue](https://github.com/azat-io/eslint-plugin-perfectionist/issues) that reports the same bug to avoid creating a duplicate.

### PR DESCRIPTION
**Pull request by hugop95**
Fixes #386

### Description

Today, users have the possibility to sort numeric enums. However, numeric enums might not directly be written with literal numeric values.

```ts
enum Period {
  A = 60,
  B = 60 * 60 * 24 * 31 // About 1 month in seconds 
}
``` 

This PR handles numeric operations on enum values to deduce their value, and sort them appropriately.

### What is the purpose of this pull request? 

- [x] Bug fix
- [x] New Feature

### CODE REVIEW

**Comment by OlivierZal** [comment]
That's not the topic, but I was wondering when reading your PR whether a `sort-operators` would be use useful, but I guess not since the operator ordering often reflects a logic that a rule cannot catch...

**Comment by OlivierZal** [comment]
@hugop95, looks good to me, I've just added a small comment.

That's not the topic, but I was wondering when reading your PR whether a sort-operators would be use useful, but I guess not since the operator ordering (with its members) often reflects a logic that a systematic rule cannot catch...

File: rules/sort-enums.ts [hunk 1]
```diff
@@ -281,3 +295,66 @@ export default createEslintRule<Options, MESSAGE_ID>({
...
+      return left + right
+    case '-':
+      return left - right
+    case '*':
+      return left * right
+    case '/':
+      return left / right
+    case '%':
+      return left % right
+    default:
```
[the remark this answers is not in the document]
Comment (hugop95) [reply rules/sort-enums.ts:357]: @OlivierZal Good idea!

**Comment by hugop95** [comment]
@OlivierZal 

> That's not the topic, but I was wondering when reading your PR whether a sort-operators would be use useful, but I guess not since the operator ordering (with its members) often reflects a logic that a systematic rule cannot catch...

I don't think users are actively looking to order operation members in an alphabetical/numerical way. Logic readability is more important as you say.

File: rules/sort-enums.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (hugop95) [reply rules/sort-enums.ts:368]: @OlivierZal Operators added: `>>`, `<<`, `|`, `&`.

File: rules/sort-enums.ts [hunk 2]
```diff
@@ -281,3 +295,77 @@ export default createEslintRule<Options, MESSAGE_ID>({
...
+    case '*':
+      return left * right
+    case '/':
+      return left / right
+    case '%':
+      return left % right
+    case '|':
+      return left | right
+    case '&':
+      return left | right
```
Comment (OlivierZal) [inline rules/sort-enums.ts:366]: &

**Review by OlivierZal — CHANGES_REQUESTED** [review]

File: rules/sort-enums.ts (hunk 2 above)
Comment (hugop95) [reply rules/sort-enums.ts]: Oops! Fixed. fa41159387aa605412d9cac43c96fa00b6abc248

File: rules/sort-enums.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (OlivierZal) [reply rules/sort-enums.ts:368]: Great! And what about the following? `^` (XOR), `~^` / `^~` (XNOR)

[the remark this answers is not in the document]
Comment (hugop95) [reply rules/sort-enums.ts:370]: @OlivierZal `^` added in 75df1b1d6808d7445684b980958ddcae4dbc5ab9. `^~` is handled through `^` and `~` directly.

**Review by azat-io — APPROVED** [review]

**Comment by azat-io** [comment]
Thanks!

**Pull request merged**